common:
    lid_topic:  "points_raw"    # "/hesai/pandar"
    imu_topic:  "imu_raw"       # "imu/data"
    time_sync_en_: false         # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
                                  # This param will take effect no matter what time_sync_en_ is. So if the time offset is not known exactly, please set as 0.0

preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 16
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    timestamp_unit: 2            # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 4                     #  视野近处盲区距离, 过大会影响对地面的匹配约束，过小会引人机器人本体
    max_range: 80

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    180
    det_range:     100.0          # 用以判断机器人本体位置与地图cube边界的距离，小于1.5倍该阈值则更新ikdtree
    extrinsic_est_en_:  false      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_T: [ 0.00, 0.0, 0.0]
    extrinsic_R: [ 0, -1, 0, 
                   1, 0, 0, 
                   0, 0, 1]

publish:
    path_en_:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: false       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en_: true
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
